crates/api/src/post/block.rs

```diff
@@ -0,0 +1,33 @@
+use activitypub_federation::config::Data;
+use actix_web::web::Json;
+use lemmy_api_common::{context::LemmyContext, post::BlockKeywordForPost, SuccessResponse};
+use lemmy_db_schema::source::user_post_keyword_block::UserPostKeywordBlock;
+use lemmy_db_views::structs::LocalUserView;
+use lemmy_utils::error::{LemmyErrorType, LemmyResult};
+
+pub async fn user_block_keyword_for_posts(
+  data: Json<BlockKeywordForPost>,
+  context: Data<LemmyContext>,
+  local_user_view: LocalUserView,
+) -> LemmyResult<Json<SuccessResponse>> {
+  for keyword in &data.keywords_to_block {
+    let trimmed = keyword.trim();
+    let length = trimmed.len();
+    if length < 3 {
+      return Err(LemmyErrorType::BlockKeywordToShort.into());
+    } else if length > 50 {
+      return Err(LemmyErrorType::BlockKeywordToLong.into());
+    }
+  }
+  if data.keywords_to_block.len() >= 15 {
+    Err(LemmyErrorType::BlockKeywordLimitReached)?;
+  }
```

Comment:
This is how length is checked in other places, and this will also change the limit to 1000 which is good. The newly created error type must also be removed.

```suggestion
  if data.keywords_to_block.len() > MAX_API_PARAM_ELEMENTS {
    Err(LemmyErrorType::TooManyItems)?;
  }
```